keylime/registrar_common.py

```diff
@@ -266,6 +266,24 @@ def do_POST(self) -> None:
 
             ekcert = json_body["ekcert"]
             aik_tpm = json_body["aik_tpm"]
+            iak_tpm = ""
+            idevid_tpm = ""
+            # Check if IDevID and IAK are received
+            # If not it is reported with logger info, but this behaviour will change when configuration requirements are added
+            idevid_received = False
+            try:
+                iak_tpm = json_body["iak_tpm"]
+                idevid_tpm = json_body["idevid_tpm"]
+                iak_attest = base64.b64decode(json_body["iak_attest"])
+                iak_sign_raw = json_body["iak_sign"]
+                iak_sign = base64.b64decode(iak_sign_raw)
+                if idevid_tpm == "AA==":
```

Comment:
Why is the separate check required? If there is no idevid this field just should be None

Reply:
The difference is if an old agent is being used this field doesn't exist, but if a new version is being used but idevid is disabled this field is '0'. I can change this from the agent side and then remove this check if that would be preferable.

Reply:
I have just committed a change to the agent to remove sending the empty fields so I have removed the checks in the newest commit.